router/broker.go

```diff
@@ -88,8 +118,14 @@ func newBroker(logger stdlog.StdLog, strictURI, allowDisclose, debug bool, publi
 		debug:         debug,
 		filterFactory: publishFilter,
 	}
+	err := b.PreInitEventHistoryTopics(evntCfgs)
+	// if broker fails initialize event history store we just log it,
+	// the broker itself will continue to operate but without event history store
+	if err != nil {
+		return nil, errors.New("error parsing/initializing event history config")
```

Comment:
I would try to retain the original error, as you otherwise have no idea as to what went wrong. In my earlier suggestion I had something like this in mind:

``` go
type configError {
    Err error
}

func (e configError) Error() string {
    fmt.Sprintf("configuration error: %v", e.Err)
}

func (e configError) Unwrap() error {
    return e.Err
}
```

This lets you do:

``` go
if err != nil {
    return nil, configError{Err: err}
}
```

and let the users have a readable error message and the programmers a way to match on the wrapped error.

You can also avoid the extra type and use `fmt.Errorf("... %w", err) with similar effect, if the nature of the wrapper type is not relevant.

Reply:
Thnx! Good idea!